code/game/objects/machinery/computer/nuke_disk_generator.dm

```diff
@@ -0,0 +1,169 @@
+		"Establishing link to offsite mainframe- Link established",
+		"WARNING, DIRECTORY CORRUPTED, running search algorithms- nuke_fission_timing.exe found",
+		"Invalid credentials, upgrading permissions through TGMC military override- Permissions upgraded, nuke_fission_timing.exe available",
+		"Downloading nuke_fission_timing.exe to removable storage- nuke_fission_timing.exe downloaded to floppy disk, have a nice day"
+	)
+
+/obj/machinery/computer/nuke_disk_generator/Initialize()
+	. = ..()
+
+	if(!disk_type)
+		WARNING("disk_type is required to be set before init")
+		return INITIALIZE_HINT_QDEL
+
+	GLOB.nuke_disk_generators += src
+
+
+/obj/machinery/computer/nuke_disk_generator/Destroy()
+	GLOB.nuke_disk_generators -= src
+	return ..()
+
+
+/obj/machinery/computer/nuke_disk_generator/process()
+	. = ..()
+	if(. || !current_timer)
+		return
+
+	deltimer(current_timer)
+	current_timer = null
+	visible_message("<b>[src]</b> shuts down as it loses power. Any running programs will now exit")
+
+
+/obj/machinery/computer/nuke_disk_generator/attackby(obj/item/I, mob/living/user, params)
+	return attack_hand(user)
+
+/obj/machinery/computer/nuke_disk_generator/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+
+	interact(user)
+
+/obj/machinery/computer/nuke_disk_generator/interact(mob/user)
+	user.set_interaction(src)
+	var/dat = ""
+	dat += "<a href='?src=[REF(src)];mach_close=computer'>Close</a><br><br>"
+	dat += "<div align='center'><a href='?src=[REF(src)];generate=1'>Run Program</a></div>"
+	dat += "<br/>"
+	dat += "<hr/>"
+	dat += "<div align='center'><h2>Status</h2></div>"
+
+	var/message = "Error"
+	if(completed_segments >= total_segments)
+		message = "Disk generated. Run program to print."
+	else if(current_timer)
+		message = "Program running"
+	else if(completed_segments == 0)
+		message = "Idle"
+	else if(completed_segments < total_segments)
+		message = "Restart required. Please rerun the program"
+	else
+		message = "Unknown"
+
+	var/progress = round((completed_segments / total_segments) * 100)
+
+	dat += "<br/><span><b>Progress</b>: [progress]%</span>"
+	dat += "<br/><span><b>Time left</b>: [current_timer ? round(timeleft(current_timer) * 0.1, 2) : 0.0]</span>"
+	dat += "<br/><span><b>Message</b>: [message]</span>"
+
+	var/flair = ""
+	for(var/i in 1 to completed_segments)
+		flair += "[technobabble[i]]<br />"
+
+	dat += "<br /><br /><span style='font-family: monospace, monospace;'>[flair]</span>"
+
+	var/datum/browser/popup = new(user, "computer", "<div align='center'>Nuke Disk Generator</div>")
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/computer/nuke_disk_generator/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return
+
+	if(href_list["generate"])
+		if(printing || current_timer)
+			to_chat(usr, "<span class='warning'>A program is already running.</span>")
+			return
+		if(reprintable)
+			printing = TRUE
+			addtimer(VARSET_CALLBACK(src, printing, FALSE), printing_time) // TODO Change to larger time

```

Comment:
No TODOs

Reply:
This is copypasted. Let's see what @psykzz wanted to do here and I'll try to get it done.

Reply:
This printing time is good, previously it had 3 SECONDS or so before i changed it to some standard number.